src/vulkan/compute_pipeline.cc

```diff
@@ -90,23 +90,25 @@ Result ComputePipeline::Compute(uint32_t x, uint32_t y, uint32_t z) {
   if (!r.IsSuccess())
     return r;
 
-  r = command_->BeginIfNotInRecording();
-  if (!r.IsSuccess())
-    return r;
+  {
+    CommandBufferGuard guard(GetCommandBuffer());
+    if (!guard.IsRecording())
+      return guard.GetResult();
 
-  BindVkDescriptorSets(pipeline_layout);
+    BindVkDescriptorSets(pipeline_layout);
 
-  r = RecordPushConstant(pipeline_layout);
-  if (!r.IsSuccess())
-    return r;
+    r = RecordPushConstant(pipeline_layout);
+    if (!r.IsSuccess())
+      return r;
```

Comment:
Example, if this recording fails, then the guard will be in non-submitted state and it will assert out.